docs/OV_Runtime_UG/auto_device_selection.md

```diff
@@ -1,332 +1,407 @@
-```
-
-The Auto-device plugin supports query device optimization capabilities in metric.
-
-| Parameter name | Parameter values |
-| --- | --- |
-| "OPTIMIZATION_CAPABILITIES" | Auto-Device capabilities |
-
-### Enumerating Devices and Selection Logic
-
-The OpenVINO Runtime API now features a dedicated methods to enumerate devices and their capabilities. See the [Hello Query Device Python Sample](../../samples/python/hello_query_device/README.md) for code.
-
-This is the example output from the sample (truncated to device names only):
-
-```python
-./hello_query_device
-
-Available devices:
-    Device: CPU
-...
-    Device: GPU.0
-...
-    Device: GPU.1
-```
-
-### Default Auto-Device Selection Logic
-
-With the 2021.4 release, the Auto-Device selects the most suitable device using the following default logic:
-
-1. Check if dGPU (discrete), iGPU (integrated) and CPU devices are available
-2. Get the precision of the input model, such as FP32
-3. According to the priority of dGPU, iGPU, and CPU (in this order), if the device supports the precision of the input network, select it as the most suitable device
-
-For example, CPU, dGPU and iGPU can support the following precision and optimization capabilities:
-
-| Device | OPTIMIZATION_CAPABILITIES |
-| --- | --- |
-| CPU | WINOGRAD FP32 FP16 INT8 BIN |
-| dGPU | FP32 BIN BATCHED_BLOB FP16 INT8 |
-| iGPU | FP32 BIN BATCHED_BLOB FP16 INT8 |
-
-* When the application uses the Auto-device to run FP16 IR on a system with CPU, dGPU and iGPU, Auto-device will offload this workload to dGPU.
-* When the application uses the Auto-device to run FP16 IR on a system with CPU and iGPU, Auto-device will offload this workload to iGPU.
-* When the application uses the Auto-device to run WINOGRAD-enabled IR on a system with CPU, dGPU and iGPU, Auto-device will offload this workload to CPU.
-
-In cases when loading the network to dGPU or iGPU fails, CPU is the fall-back choice.
-
-To show the capabilities for a specific device, query the OPTIMIZATION_CAPABILITIES metric:
-
-
-```python
-from openvino.inference_engine import IECore
-
-ie = IECore()
-ie.get_metric(device_name=device,
-              metric_name="OPTIMIZATION_CAPABILITIES")
-```
-
-### Configuring the Individual Devices and Creating the Auto-Device on Top
-
-It is possible to configure each individual device as usual and create the "AUTO" device on top:
-
-```python
-from openvino.inference_engine import IECore
-
-ie = IECore()
-net = ie.read_network(model=path_to_model)
-
-cpu_config = {}
-gpu_config = {}
-
-ie.set_config(config=cpu_config, device_name="CPU")
-ie.set_config(config=gpu_config, device_name="GPU")
-
-# Load the network to the AUTO device
-exec_net = ie.load_network(network=net, device_name="AUTO")
-```
-
-Alternatively, you can combine all the individual device settings into single config file and load it, allowing the Auto-device plugin to parse and apply it to the right devices. See the code example here:
+For C++ API
```

Comment:
Why C++/Python tabs aren't used here?

Reply:
with just one line of code the tab option seems excessive and disruptive. I does not make overall content much shorter, if at all.